### PR TITLE
windows service host for web and worker roles

### PR DESCRIPTION
**Pull request by Drewan**
Topshelf - An easy service hosting framework for building Windows services https://github.com/Topshelf/Topshelf

Set LightBlue.WorkerService or LightBlue.WebService as startup project and hit F5 to debug as a console application.

image

OR

To run the windows service open a command prompt as administrator and pass the following arguments to the LightBlue.WebService or LightBlue.WorkerService executable.
- install
- start
- stop
- uninstall

image

image

**AppVeyor** https://ci.appveyor.com/project/Lightblue/lightblue

The Appveyor setup to build, test, and package LightBlue is configured from the .yml file in the root directory.    

image

Each build produces these nuget package artifacts that need to be manually uploaded to nuget.org at this stage.

image

### CODE REVIEW

File: LightBlue.Service/AppConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+{
+    public class AppConfiguration
+    {
+        public static T Load<T>() where T : class
+        {
+            dynamic configuration = new Configuration();
+            T settings = configuration.Bind<T>();
+            return settings;
+        }
+    }
```
Comment (Drewan) [inline LightBlue.Service/AppConfiguration.cs]: Formo https://github.com/ChrisMissal/Formo reads AppSettings from app.config files

File: LightBlue.WebService/Program.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        {
+           return (int)HostFactory.Run(x =>
+           {
+               var settings = AppConfiguration.Load<WebHostSettings>();
+
+               x.Service<WebHostService>(service =>
+               {
+                   service.ConstructUsing(s =>
+                   {
+                       var hostArgs = WebHostArgs.ParseArgs(new[]
```
Comment (Drewan) [inline LightBlue.WebService/Program.cs:20]: Leveraging existing validation on WebHostArgs

File: LightBlue.WebService/Program.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+               dynamic configuration = new Configuration();
+               WebHostSettings settings = configuration.Bind<WebHostSettings>();
+
+               x.Service<WebHostService>(service =>
+               {
+                   service.ConstructUsing(s =>
+                   {
+                       var hostArgs = WebHostArgs.ParseArgs(new[]
+                       {
+                           "-a:" + settings.Assembly,
```
Comment (PeteNeedle) [inline LightBlue.WebService/Program.cs:22]: Will this only work if the code is kept in C:\Source\LightBlue\ ?

Comment (Drewan) [reply LightBlue.WebService/Program.cs:22]: I'm pretty sure it uses the -a: assembly (path to role assembly with all dlls)  and -c: configuration path (path to .cscfg) so it doesn't need to make any assumptions on other file paths.

File: LightBlue.WebHost/LightBlue.WebHost.nuspec [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<package >
+	<metadata>
+		<id>$id$</id>
+		<version>$version$</version>
+		<title>$id$</title>
+		<authors>Colin David Scott</authors>
```
Comment (eric-winkler) [inline LightBlue.WebHost/LightBlue.WebHost.nuspec]: Should we use `$author$` for authors and owners?

File: LightBlue.Autofac.nuspec [hunk 5]
```diff
@@ -5,7 +5,7 @@
     <version>1.1.21-beta</version>
 	<title>LightBlue.Autofac</title>
     <authors>Colin David Scott</authors>
-    <owners>Colin David Scott</owners>
```
Comment (eric-winkler) [inline LightBlue.Autofac.nuspec:8]: This one is inconsistent?

**Review by eric-winkler — APPROVED** [review]

**Pull request merged**